pkg/cmd/cli/cmd/expose.go

```diff
@@ -119,8 +119,13 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 			supportsTCP := false
 			for _, port := range svc.Spec.Ports {
 				if port.Protocol == kapi.ProtocolTCP {
-					// Pass service target port as the route port
-					cmd.Flags().Set("port", port.TargetPort.String())
+					if len(port.Name) > 0 {
+						// Pass service port name as the route target port, if it is named
+						cmd.Flags().Set("port", port.Name)
+					} else {
+						// Otherwise, omit the targetPort from the route and let it round-robin the one service port
+						cmd.Flags().Set("omit-target-port", "true")
```

Comment:
ugly, I don't see a better way to explicitly tell the route generator we want it to leave out the port